packages/react-component-library/src/components/SelectBase/SelectLayout.tsx

```diff
@@ -59,66 +67,81 @@ export const SelectLayout: React.FC<SelectLayoutProps> = ({
             >
-              {label}
-            </StyledLabel>
-            <StyledInput
-              $hasLabel
-              data-testid="select-input"
-              disabled={isDisabled}
-              value={value}
-              onMouseEnter={() => {
-                if (!isDisabled) {
-                  setHasHover(true)
-                }
-              }}
-              onMouseLeave={() => setHasHover(false)}
-              {...rest}
-              {...inputProps}
-              aria-labelledby={labelId}
-              id={id}
-            />
-          </StyledInputWrapper>
-          <StyledInlineButtons>
-            {hasSelectedItem && (
-              <ClearButton
+              <StyledLabel
+                $hasContent={hasSelectedItem}
+                $hasFocus={hasLabelFocus}
+                htmlFor={id}
+                id={labelId}
+                data-testid="select-label"
+              >
+                {label}
+              </StyledLabel>
+              <StyledInput
+                ref={mergeRefs([setFloatingBoxTarget, ref])}
+                $hasLabel
+                data-testid="select-input"
+                disabled={isDisabled}
+                value={value}
+                onMouseEnter={() => {
+                  if (!isDisabled) {
+                    setHasHover(true)
+                  }
+                }}
+                onMouseLeave={() => setHasHover(false)}
+                {...rest}
+                {...inputPropsRest}
+                aria-labelledby={labelId}
+                id={id}
+              />
+            </StyledInputWrapper>
+            <StyledInlineButtons>
+              {hasSelectedItem && (
+                <ClearButton
+                  isDisabled={isDisabled}
+                  onClick={onClearButtonClick}
+                />
+              )}
+              <ArrowButton
+                hasHover={hasHover}
                 isDisabled={isDisabled}
-                onClick={onClearButtonClick}
+                isOpen={isOpen}
+                {...toggleButtonProps}
               />
-            )}
-            <ArrowButton
-              hasHover={hasHover}
-              isDisabled={isDisabled}
-              isOpen={isOpen}
-              {...toggleButtonProps}
-            />
-          </StyledInlineButtons>
-        </StyledOuterWrapper>
-      </StyledTextInput>
-      <StyledOptionsWrapper $isVisible={isOpen}>
-        <StyledOptions {...menuProps}>{children}</StyledOptions>
-      </StyledOptionsWrapper>
-    </StyledSelect>
+            </StyledInlineButtons>
+          </StyledOuterWrapper>
+        </StyledTextInput>
+        <StyledOptionsWrapper $isVisible={isOpen}>
+          <StyledOptions {...menuProps}>{children}</StyledOptions>
+        </StyledOptionsWrapper>
+      </StyledSelect>
+      <FloatingBox
+        placement="bottom"
+        scheme="dark"
+        isVisible={
+          hasHover && isEllipsisActive(floatingBoxTarget as HTMLInputElement)
```

Comment:
`useStatefulRef<HTMLInputElement>()` should work above I think (and get rid of the need for `as HTMLInputElement`).